s3/src/main/scala/akka/stream/alpakka/s3/impl/HttpRequests.scala

```diff
@@ -53,20 +54,21 @@ private[alpakka] object HttpRequests {
     } yield {
       s3Request(
         upload.s3Location,
+        region,
         HttpMethods.POST,
         _.withQuery(Query("uploadId" -> upload.uploadId))
       ).withEntity(entity)
     }
   }
 
-  def requestHost(s3Location: S3Location)(implicit conf: S3Settings): Uri.Host =
+  def requestHost(s3Location: S3Location, region: String)(implicit conf: S3Settings): Uri.Host =
     conf.proxy match {
-      case None => Uri.Host(s"${s3Location.bucket}.s3.amazonaws.com")
+      case None => Uri.Host(s"${s3Location.bucket}.s3-${region}.amazonaws.com")
```

Comment:
This change seems like unnecessary overhead as this changes the library where a user HAS to provide a region. When interacting with S3 using the {bucket}.s3.amazon.com you don't need to provide a region as buckets are already pinned to a region as all S3 bucket names MUST be unique. http://docs.aws.amazon.com/AmazonS3/latest/gsg/CreatingABucket.html